Protobuf version matching error

Hello,
First, thank you for creating the tensorflow_cc software module; so far, this appears to be the most succinct and functional solution for setting up the _hard-to-find-decent-documentation-for_  software known as TensorFlow's C APIs.

Here is my problem:
I am attempting to use the shared tensorflow_cc library version within
a custom docker configuration in Ubuntu 16.04.  I have used the
"ubuntu" dockerfile in the "Dockerfiles" folder of tensorflow_cc as a reference for setting
up the appropriate dependencies.  And the example application builds
and runs just fine after I complete the process specified in your README.md.

However, the challenge comes when I attempt to make (after cmake'ing) my own project
which is attempting to leverage the C-API of TensorFlow, linking to your libraries.
Specifically, what happens is that "make" outputs many instances of the same error message:

**#error This file was generated by a newer version of protoc which is
#error incompatible with your Protocol Buffer headers.  Please update
#error your headers.**

with regards to various protobuf compiler generated files, such as
`/usr/local/include/tensorflow/bazel-genfiles/tensorflow/core/framework/tensor.pb.h.`
And these files are associated with TensorFlow-related header files included in my
project, such as `/usr/local/include/tensorflow/tensorflow/cc/ops/const_op.h.`

I have verified that the version of Protobuf installed on my system:
3.5.0, matches that employed by tensorflow_cc (via much searching and
emitting of messages from the common.h files). (Is that version correct?)
Furthermore, I have installed and built the Google Protobuf library 3.5.0 from source to ensure the correct linkings.  However, there seems to be another, hidden issue causing the protobuf mismatch error.
In my software, there unfortunately are a few other libraries required to exist
which have their own versions of protobuf.  I have worked on
severing the ties to these wrong versions, but I am not 100% that I have done
the right thing in this regard utilizing my CMakeLists.txt file.  Maybe you can help?  Specifically,
how do I link against the correct tensorflow header files and/or the correct
protobuf compiler-generated files in my project?  Your example application
shows how to set up the `target_link_libraries() `command, but not 
`include_directories()` or` target_include_directories()`, which I feel like
are still needed (maybe not?).

Here is a sample of the related components of my CMakeLists.txt:

```
find_package(TensorflowCC COMPONENTS Shared)
if(TensorflowCC_Shared_FOUND)
    message(STATUS "FOUND TensorflowCC_Shared library")
else(TensorflowCC_Shared_FOUND)
    message(STATUS "DID NOT FIND TensorflowCC_Shared library")
endif(TensorflowCC_Shared_FOUND)

find_package(Protobuf REQUIRED)
if(Protobuf_FOUND)
    message(STATUS "PROTOBUF_INCLUDE_DIRS == ${PROTOBUF_INCLUDE_DIRS}")	# this location looks fine
    message(STATUS "PROTOBUF_LIBRARIES    == ${PROTOBUF_LIBRARIES}")    # this location looks fine
else(Protobuf_FOUND)
    message(STATUS "Find_package(protobuf) failed")
endif(Protobuf_FOUND)

include_directories(${PROTOBUF_INCLUDE_DIRS})
include_directories(${CMAKE_CURRENT_BINARY_DIR})

add_executable( MY_TENSORFLOW_CPP_MODULE
                my_tensorflow_module.cpp
                ${PROTO_SRCS}
                ${PROTO_HDRS} )

target_include_directories( MY_TENSORFLOW_CPP_MODULE PRIVATE
                            tensorflow_cc_shared
                            ${PROTOBUF_INCLUDE_DIRS} )

target_link_libraries(  MY_TENSORFLOW_CPP_MODULE PRIVATE
                            ${PROTOBUF_LIBRARIES}
                            TensorflowCC::Shared 

install(TARGETS MY_TENSORFLOW_CPP_MODULE DESTINATION bin)

```
Any ideas on what I might be missing or configuring incorrectly here?

Many thanks for your help!
-Robert

Thank you!  Unfortunately, I get the following error when I attempt the above fix:

```
CMake Error at CMakeLists.txt:48 (message):
  SYSTEM_PROTOBUF is only supported with TENSORFLOW_STATIC.
```
And I need the shared version of the library for GPU support and more features.
I suppose I could try the static library for now, but I'm not sure if that will fully
meet my needs.

I think there was a simple check for shared vs. static with respect to the
SYSTEM_PROTOBUF parameter in one of the underlying CMake files...
if this were disabled, could this feature work?  Or is it a deeper problem.

--- a/tensorflow_cc/CMakeLists.txt
+++ b/tensorflow_cc/CMakeLists.txt
@@ -45,9 +45,6 @@ endif()
 # Define Shared Tensorflow Interface
 # ----------------------------------
 if(SYSTEM_PROTOBUF)
-  if(NOT TENSORFLOW_STATIC)
-    message(FATAL_ERROR "SYSTEM_PROTOBUF is only supported with TENSORFLOW_STATIC.")
-  endif()
   find_package(Protobuf REQUIRED)
 endif()
 

